### PR TITLE
[FLINK-22782][docs] Remove legacy planner from Chinese docs

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*This is another step in getting rid of the legacy planner. The Chinese docs are still not perfect but better than before with this PR. The Chinese docs should be done in a followup issue, I only dropped the legacy planner page there.*

## Brief change log

  - Remove reference of `useBlink/LegacyPlanner`
  - Remove `DataSet`
  - Remove legacy planner mentioning

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/content.zh/docs/dev/table/common.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ import org.apache.flink.table.api.EnvironmentSettings
 import org.apache.flink.table.api.bridge.scala.StreamTableEnvironment
 
 val bsEnv = StreamExecutionEnvironment.getExecutionEnvironment
```
Comment (twalthr) [inline docs/content.zh/docs/dev/table/common.md:226]: remove `bs` and `b_s_settings` etc. here and below? I think the "b" stood for Blink in the past?

File: docs/content.zh/docs/dev/table/common.md [hunk 2]
```diff
@@ -267,15 +267,15 @@ from pyflink.datastream import StreamExecutionEnvironment
 from pyflink.table import StreamTableEnvironment, EnvironmentSettings
 
 b_s_env = StreamExecutionEnvironment.get_execution_environment()
-b_s_settings = EnvironmentSettings.new_instance().use_blink_planner().in_streaming_mode().build()
+b_s_settings = EnvironmentSettings.new_instance().in_streaming_mode().build()
 b_s_t_env = StreamTableEnvironment.create(b_s_env, environment_settings=b_s_settings)
 
 # ******************
 # BLINK BATCH QUERY
```
Comment (twalthr) [inline docs/content.zh/docs/dev/table/common.md:274]: update?

File: docs/content.zh/docs/dev/table/tableApi.md [hunk 3]
```diff
@@ -118,7 +117,7 @@ from pyflink.table import *
 
 # environment configuration
 t_env = BatchTableEnvironment.create(
```
Comment (twalthr) [inline docs/content.zh/docs/dev/table/tableApi.md:119]: update usages of `BatchTableEnvironment`?

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks for the PR @SteNicholas. Could you perform another search in the Chinese dcos for occurences of `DataSet`, `ExecutionEnvironment`, `BatchTableEnvironment`. It seems there are still a lot of mentionings. Such as in `zh/docs/dev/table/common/ ` with `fsSettings` or `useAnyPlanner`. The English docs were already updated.

**Review by wuchong — COMMENTED** [review]
1. should we remove all the "blink" keyword in the docs? cc @twalthr 
2. there are still many "dataset" used in `docs/content.zh/docs/dev/table/functions/udfs.md`.

**Comment by twalthr** [comment]
Yes, the removal of "blink" keyword will be another subtask. But I think the PR is good for now. Let's do all of this incrementally. But we could remove `dataset` in `udfs` already. What do you think @SteNicholas?

**Comment by SteNicholas** [comment]
@wuchong @twalthr , thanks for detailed review. I have updated the `docs/content.zh/docs/dev/table/functions/udfs.md` for `dataset`. Please help to review again.

**Review by twalthr — APPROVED** [review]
Thanks @SteNicholas. LGTM I will merge this now. We will do another pass over the docs at a later stage.

**Pull request closed without merging**